crates/accelerate/src/commutation_cancellation.rs

```diff
@@ -0,0 +1,291 @@
+                                                qubits: smallvec![wire],
+                                                com_set_index: com_set_idx,
+                                                second_index: None,
+                                            })
+                                            .or_insert_with(Vec::new)
+                                            .push(*node);
+                                    }
+                                    // Don't deal with Y rotation, because Y rotation doesn't commute with
+                                    // CNOT, so it should be dealt with by optimized1qgate pass
+                                    if num_qargs == 2usize
+                                        && dag.get_qargs(op.qubits).first().unwrap() == &wire
+                                    {
+                                        let second_qarg = dag.get_qargs(op.qubits)[1];
+                                        cancellation_sets
+                                            .entry(CancellationSetKey {
+                                                gate: GateOrRotation::Gate(op_gate),
+                                                qubits: smallvec![wire, second_qarg],
+                                                com_set_index: com_set_idx,
+                                                second_index: node_indices
+                                                    .get(&(*node, Wire::Qubit(second_qarg)))
+                                                    .copied(),
+                                            })
+                                            .or_insert_with(Vec::new)
+                                            .push(*node);
+                                    }
+                                }
+                            }
+                        })
+                    }
+                })
+        }
+    });
+
+    for (cancel_key, cancel_set) in &cancellation_sets {
+        if cancel_set.len() > 1 {
+            if let GateOrRotation::Gate(g) = cancel_key.gate {
+                if SUPPORTED_GATES.contains(&g) {
+                    for &c_node in &cancel_set[0..(cancel_set.len() / 2) * 2] {
+                        dag.remove_op_node(c_node);
+                    }
+                }
+            }
+            if matches!(cancel_key.gate, GateOrRotation::ZRotation) && z_var_gate.is_none() {
+                continue;
+            }
+            if matches!(cancel_key.gate, GateOrRotation::ZRotation)
+                || matches!(cancel_key.gate, GateOrRotation::XRotation)
+            {
+                let run_op = match &dag.dag[*cancel_set.first().unwrap()] {
+                    NodeType::Operation(instr) => instr,
+                    _ => panic!("Unexpected type in commutation set run."),
+                };
+
+                let run_qarg = dag.get_qargs(run_op.qubits).first().unwrap();
```

Comment:
It's a slice so you can just get this with an index:
```suggestion
                let run_qarg = dag.get_qargs(run_op.qubits)[0];
```